File: src/main/java/com/pepebot/config/WebClientConfig.java
package com.pepebot.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.client.WebClient;

@Configuration
public class WebClientConfig {

	@Bean
	WebClient krakenWebClient(WebClient.Builder builder, KrakenProperties properties) {
		return builder.baseUrl(properties.getBaseUrl()).build();
	}
}

File: src/main/kotlin/com/pepebot/config/WebClientConfig.kt
package com.pepebot.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.client.WebClient

@Configuration(proxyBeanMethods = false)
class WebClientConfig {
    @Bean
    fun krakenWebClient(builder: WebClient.Builder, properties: KrakenProperties): WebClient {
        return builder.baseUrl(properties.baseUrl).build()
    }
}
